### DISCUSSION TITLE
What does `yosys -import` do?

### DISCUSSION THREAD

**Discussion by xobs** [The-OpenROAD-Project/OpenLane#1727] [Q&A] [answered]
In the synthesis script, the flow runs `yosys -import`:

https://github.com/The-OpenROAD-Project/OpenLane/blob/cb634fd5f670db4a12e36075c642dae54150955c/scripts/yosys/synth.tcl#L14

However, this appears to be neither a valid yosys command, nor a valid yosys argument. What is this command and where does it come from?

**Comment by kareefardi** [comment, marked as answer]
@xobs Without it we can't use yosys builtin tcl commands, as far as I understand. From yosys docs
>The tcl command 'yosys -import' can be used to import all yosys
commands directly as tcl commands to the tcl shell. Yosys commands
'proc' and 'rename' are wrapped to tcl commands 'procs' and 'renames'
in order to avoid a name collision with the built in commands.

**Reply by xobs** [reply]
I see, thank you.

Normally, I've used Yosys in interactive mode, and I've therefore used the custom Yosys script language.

OpenLane uses Yosys in tcl mode, which behaves slightly differently. Notably, commands aren't available by default as TCL verbs. The TCL function `yosys -import` re-imports all Yosys commands into TCL land.

This needs to be done any time additional verbs are added, such as when a plugin is loaded.

**Reply by kareefardi** [reply]
Yes. Thanks for better explaining it.
